### PR TITLE
fix removeLayers when cluster is not on the map

### LINKED ISSUE #545: removeLayers not working when cluster is not on map
I've prepared a small [fiddle](http://jsfiddle.net/Eschon/92aLfywq/) to show my problem.
I added a cluster to the map and added a marker to the cluster.
Then I removed the cluster from the map. While the cluster isn't on the map I try to remove the marker using the removeLayers function. When I add the cluster back to the map the marker is still there.

I know that I could just use removeLayer for this small example and it works fine, but for my real use case I have a lot of markers and looping through all of them and removing then one at a time isn't good for performance.

### PR DESCRIPTION
**Pull request by Eschon**
Fixes #545

### CODE REVIEW

**Comment by Eschon** [comment]
I don't know why the build failed.
`jake test` runs without problems for me.

**Comment by danzel** [comment]
Re-ran the build, worked now.

This change doesn't seem quite right though, we do the array splice two lines up.

Also a test that requires this fix would be great.

**Comment by Eschon** [comment]
Ah you're right. I just copied the code that adds the makrer to _needsRemoving from removeLayer and overlooked the duplicate _arraySplice.

Using this fix removes the bug from my [fiddle](http://jsfiddle.net/Eschon/92aLfywq/4/).

I have no experience writing test but I'll try to add one.

**Comment by Eschon** [comment]
Now I also added a test for the fix I hope it is ok.

File: src/MarkerClusterGroup.js [hunk 1]
```diff
@@ -260,6 +260,9 @@ L.MarkerClusterGroup = L.FeatureGroup.extend({
 				m = layersArray[i];
 				this._arraySplice(this._needsClustering, m);
 				npg.removeLayer(m);
+				if (this.hasLayer(m)) {
```
Comment (danzel) [inline src/MarkerClusterGroup.js]: Add a comment on this detailing why it is done please, it looks a bit weird when you look at it standalone.

Comment (Eschon) [inline src/MarkerClusterGroup.js]: To be honest I'm not sure if we really need that check. The important part is that the layer is added to the _needsRemoving array. 
The check is there when removing a single Layer so I added it here too.

**Pull request merged**